pkg/apis/core/v1/helper/helpers.go

```
@@ -287,28 +289,36 @@ func AddOrUpdateTolerationInPodSpec(spec *v1.PodSpec, toleration *v1.Toleration)
 }
 
 // GetMatchingTolerations returns true and list of Tolerations matching all Taints if all are tolerated, or false otherwise.
-func GetMatchingTolerations(taints []v1.Taint, tolerations []v1.Toleration) (bool, []v1.Toleration) {
+func GetMatchingTolerations(taints []v1.Taint, tolerations []v1.Toleration) (bool, []v1.Toleration, error) {
+	var err errorsutil.Aggregate
 	if len(taints) == 0 {
-		return true, []v1.Toleration{}
+		return true, []v1.Toleration{}, nil
 	}
 	if len(tolerations) == 0 && len(taints) > 0 {
-		return false, []v1.Toleration{}
+		return false, []v1.Toleration{}, nil
 	}
 	result := []v1.Toleration{}
+
+	ttErrs := []error{}
 	for i := range taints {
 		tolerated := false
 		for j := range tolerations {
-			if tolerations[j].ToleratesTaint(&taints[i]) {
+			isTolerated, err := tolerations[j].ToleratesTaint(&taints[i])
+			if err != nil {
+				ttErrs = append(ttErrs, err)
+			}
+			if isTolerated {
```

Maybe just return an error for the first wrong toleration?